pkg/k8s/constants.go

```diff
@@ -157,6 +157,14 @@ var (
 		// KCC no longer supports GameServicesRealm CRD as of v1.101.0.
 		"GameServicesRealm": true,
 	}
+
+	// ObservedStateAllowlist is a map containing a list of resources that
```

Comment:
Could you add a comment to explain this is only introduced for TF based resources?

Reply:
Good point! I actually forgot about DCL-based resources, so I made the allowlist available for DCL-based resources as well.